### ISSUE TITLE
test failure due to overly strict clang version string check

### ISSUE THREAD

**Issue by anbe42** [KhronosGroup/SPIRV-LLVM-Translator#2176]
While packaging spirv-llvm-translator 17.0.0 for Debian, I ran into a test failure because `String [[#CV:]] "{{.*}}clang version [[#]].0.0` does not match `12 String 51 "Debian clang version 17.0.2 (1~exp1)"`:
```
FAIL: LLVM_SPIRV :: DebugInfo/NonSemantic/DebugFunction.cl (77 of 708)
******************** TEST 'LLVM_SPIRV :: DebugInfo/NonSemantic/DebugFunction.cl' FAILED ********************
Script:
--
: 'RUN: at line 7';   /usr/lib/llvm-17/bin/clang -cc1 -internal-isystem /usr/lib/llvm-17/lib/clang/17/include -nostdsysteminc /build/spirv-llvm-translator-17-17.0.0/test/DebugInfo/NonSemantic/DebugFunction.cl -cl-std=clc++ -emit-llvm-bc -triple spir -debug-i
nfo-kind=line-tables-only -O0 -o /build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.bc
: 'RUN: at line 8';   /build/spirv-llvm-translator-17-17.0.0/build/tools/llvm-spirv/llvm-spirv /build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.bc --spirv-debug-info-version=nonsemantic-shader-10
0 -o /build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.spv
: 'RUN: at line 9';   /build/spirv-llvm-translator-17-17.0.0/build/tools/llvm-spirv/llvm-spirv /build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.spv -to-text -o /build/spirv-llvm-translator-17-17.
0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.spt
: 'RUN: at line 10';   /usr/lib/llvm-17/bin/FileCheck /build/spirv-llvm-translator-17-17.0.0/test/DebugInfo/NonSemantic/DebugFunction.cl --input-file /build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.
tmp.spt  --check-prefix=CHECK-SPIRV
: 'RUN: at line 12';   /build/spirv-llvm-translator-17-17.0.0/build/tools/llvm-spirv/llvm-spirv /build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.bc --spirv-debug-info-version=nonsemantic-shader-2
00 -o /build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.spv
: 'RUN: at line 13';   /build/spirv-llvm-translator-17-17.0.0/build/tools/llvm-spirv/llvm-spirv /build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.spv -to-text -o /build/spirv-llvm-translator-17-17
.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.spt
: 'RUN: at line 14';   /usr/lib/llvm-17/bin/FileCheck /build/spirv-llvm-translator-17-17.0.0/test/DebugInfo/NonSemantic/DebugFunction.cl --input-file /build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.
tmp.spt  --check-prefix=CHECK-SPIRV
: 'RUN: at line 16';   /build/spirv-llvm-translator-17-17.0.0/build/tools/llvm-spirv/llvm-spirv -r -emit-opaque-pointers /build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.spv -o - | /usr/lib/llvm-
17/bin/llvm-dis -o - | /usr/lib/llvm-17/bin/FileCheck /build/spirv-llvm-translator-17-17.0.0/test/DebugInfo/NonSemantic/DebugFunction.cl --check-prefix=CHECK-LLVM
--
Exit Code: 1

Command Output (stderr):
--
/build/spirv-llvm-translator-17-17.0.0/test/DebugInfo/NonSemantic/DebugFunction.cl:28:21: error: CHECK-SPIRV-DAG: expected string not found in input
// CHECK-SPIRV-DAG: String [[#CV:]] "{{.*}}clang version [[#]].0.0
                    ^
/build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.spt:1:1: note: scanning from here
119734787 65536 393230 73 0 
^
/build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.spt:17:4: note: possible intended match here
12 String 51 "Debian clang version 17.0.2 (1~exp1)" 
   ^

Input file: /build/spirv-llvm-translator-17-17.0.0/build/test/test_output/DebugInfo/NonSemantic/Output/DebugFunction.cl.tmp.spt
Check file: /build/spirv-llvm-translator-17-17.0.0/test/DebugInfo/NonSemantic/DebugFunction.cl

-dump-input=help explains the following input dump.
...
```

Commit 60c0cb0078658e0bc33d93724ffbe70ed4be6c51 (which only exists in branch llvm_release_140) fixes this. It should probably be cherry-picked into main and backported to the other release branches.

**Comment by MrSidims** [comment]
Thanks for the report! I'm addressing the issue

**Comment by svenvh** [comment]
This has been fixed by #2194
